x-pack/plugins/snapshot_restore/public/app/sections/repository_add/repository_add.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useEffect, useState } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+
+import { API_BASE_PATH, REPOSITORY_TYPES } from '../../../../common/constants';
+import { Repository } from '../../../../common/types';
+
+import { RepositoryForm, SectionError } from '../../components';
+import { BASE_PATH, getHomeBreadcrumb, getRepositoryAddBreadcrumb, Section } from '../../constants';
+import { useAppDependencies } from '../../index';
+import { sendRequest } from '../../services/http';
+
+import { EuiPageBody, EuiPageContent, EuiSpacer, EuiTitle } from '@elastic/eui';
+
+export const RepositoryAdd: React.FunctionComponent<RouteComponentProps> = ({ history }) => {
+  const {
+    core: { i18n, http, chrome },
+    plugins: { management },
+  } = useAppDependencies();
+  const { FormattedMessage } = i18n;
+  const section = 'repositories' as Section;
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [saveError, setSaveError] = useState<null | any>(null);
+
+  // Set breadcrumb
+  useEffect(() => {
+    chrome.breadcrumbs.set([
+      management.constants.BREADCRUMB,
+      getHomeBreadcrumb(i18n.translate),
+      getRepositoryAddBreadcrumb(i18n.translate),
+    ]);
+  }, []);
+
+  const onSave = async (newRepository: Repository) => {
+    setIsSaving(true);
+    setSaveError(null);
+    const { name } = newRepository;
+    const { error } = await sendRequest({
```

Comment:
Here too it'd be cleaner a request creator:  `await saveRepository(newRepository)`